### PR TITLE
LPD-26543 Adapt function validation rule engine to support system objects

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-26543](https://liferay.atlassian.net/browse/LPD-26543)

### CODE REVIEW

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
@devpedrotavares please try to write a test for this use case

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -436,8 +436,7 @@ else if (StringUtil.startsWith(
 					null);
 			}
 			else {
-				results = objectValidationRuleEngine.execute(
-					(Map<String, Object>)variables.get("entryDTO"), null);
+				results = objectValidationRuleEngine.execute(variables, null);
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:439]: @devpedrotavares will this affect other implementations besides the `FunctionObjectValidationRuleEngineImpl`?

Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:439]: @carolmariaabb no, the other implementations will fall under one of the other clauses

**Comment by carolmariaabb** [comment]
@devpedrotavares please see https://github.com/liferay-objects/liferay-portal/pull/3383/

**Pull request closed without merging**